src/indent.ts

```diff
@@ -29,11 +31,14 @@ export function newlineAndIndent(
         if (textEditor.document.languageId === 'python') {
             const lines = textEditor.document.getText(
                 new vscode.Range(0, 0, position.line, position.character)).split("\n");
-            let indent = nextIndentationLevel(lines, tabSize);
+
+            const { nextIndentationLevel, parseOutput: { dedentNext } } = indentationInfo(lines, tabSize);
+            let indent = nextIndentationLevel;
+
             const spacesToRemove = currentLineDedentation(lines, tabSize);
             if (spacesToRemove > 0) {
                 // don't dedent the current line if we already dedented it, e.g. after a "return"
-                if (!parseLines(lines.slice(0, -1)).dedentNext) {
+                if (dedentNext) {
```

Comment:
This seems like different behavior

Reply:
Your'e right. I pushed a commit to fix it.

Reply:
I have to head to work, but I think this change effected the following kind of construct:

```python
    if True:
        1
    else:
        
```

On master, if you type

```python
    if True:
        1
        else:
```

then press `Enter`, the `else:` gets dedented and your cursor gets put in the right place. On this branch, the `else:` does not get dedented.

Reply:
Didn't see your reply before I put mine in. I'll look at it hopefully later tonight